WordPress/gutenberg
# [FEAT] Support linear gradient in cover block

## Description
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/1860

Gutenberg-mobile PR - https://github.com/wordpress-mobile/gutenberg-mobile/pull/1949

ATM we are able to set the linear gradient as a background in the cover block. Since we don't have the gradient picker on mobile yet I added a possibility to render linear-gradient if it set on the web.

Additionally, I moved the logic from ColorBackground related to the gradient to a separate file - LinearGradient.


## How has this been tested?
### Test case 1
- Add cover block on the web with **predefined** gradient as a background (it has to be linear gradient)
- Open that post on mobile
- The gradient background should be applied to the cover block on mobile

### Test case 2
- Add cover block on the web with **custom** gradient as a background (it has to be linear gradient)
- Open that post on mobile
- The custom gradient background should be applied to the cover block on mobile

### Test case 3
- Set gradient as a background as in previous test cases
- Click on the button in `BlockToolba`r and choose media to set an image as a background
- The gradient should be changed to the image you selected

### Test case 4
- Set gradient as a background as in previous test cases
- Open block settings of the cover block
- there should be only one option "Minimum height in pixel" since we don't support overlay on top of gradient

## Screenshots 
![Simulator Screen Shot - iPhone X - 2020-02-24 at 10 33 55](https://user-images.githubusercontent.com/16336501/75143450-5ae2a300-56f5-11ea-9843-70519c15e525.png)


## Types of changes





## Checklist:
